dask_pytorch/data.py

```diff
@@ -0,0 +1,97 @@
+"""
+Utilities for loading PyTorch data in distributed environments
+"""
+
+
+import tempfile
+from os.path import basename, dirname
+
+from PIL import Image
+from torch.utils.data import Dataset
+
+
+"""
+In the following, we are explicitly avoiding s3fs because it does not behave well with
+multiprocessing (which is commonly used in PyTorch dataloaders
+
+https://github.com/dask/s3fs/issues/369
+"""  # pylint: disable=pointless-string-statement
+
+
+def get_all_files(bucket: str, prefix: str, s3_client=None):
+    """
+    Get list of all files from an s3 bucket matching a certain prefix
+    """
+    import boto3  # pylint: disable=import-outside-toplevel
+
+    if s3_client is None:
+        s3_client = boto3.client("s3")
+    paginator = s3_client.get_paginator("list_objects")
+    all_files = []
+    for page in paginator.paginate(Bucket=bucket, Prefix=prefix):
+        files = [x["Key"] for x in page["Contents"]]
+        all_files.extend(files)
+    return all_files
+
+
+def read_s3_fileobj(bucket, path, fileobj):
+    """
+    read an obj from s3 (in a bucket a a certain path) to a file like object
```

Comment:
```suggestion
    read an obj from s3 to a file like object
```

Every object in S3 is "in a bucket at a certain path", so I think that parenthetical could be dropped.

I also think it would be clearer to use `bucket_name` instead of `bucket`. It's always annoying when you find a new "does-stuff-with-s3" library in the wild and they aren't clear about whether or not you're expected to add `s3://` to the bucket string you provide.